esf.tf

```diff
@@ -69,7 +74,7 @@ module "esf-lambda-function" {
     sqs_receive = {
       effect    = "Allow",
       actions   = ["sqs:ReceiveMessage", "sqs:DeleteMessage", "sqs:GetQueueAttributes"],
-      resources = [aws_sqs_queue.esf-continuing-queue.arn, aws_sqs_queue.esf-replay-queue.arn, aws_sqs_queue.esf-queue.arn]
+      resources = [aws_sqs_queue.esf-continuing-queue.arn, aws_sqs_queue.esf-replay-queue.arn, each.value.arn]
```

Comment:
this indeed is not needed. we can remove `aws_sqs_queue.esf-queue.arn` from the original version, and add a managed policy instead (https://github.com/terraform-aws-modules/terraform-aws-lambda?tab=readme-ov-file#additional-iam-policies-for-lambda-functions, https://github.com/terraform-aws-modules/terraform-aws-lambda/issues/474)

something like:
```
  attach_policies = true
  policies        = [
    "arn:aws:iam::aws:policy/AWSLambdaSQSQueueExecutionRole",
  ]
```